### PR TITLE
Move all project configuration to pyproject.toml

### LINKED ISSUE #88: Use `pyproject.toml` only
The following metadata can and should be put into the existing `pyproject.toml` file:

* `setup.cfg`
* `requirements.txt`
* `requirements-dev.txt`
* `requirements-doc.txt`

### PR DESCRIPTION
**Pull request by pont-us**
Closes #88 .

Checklist (strike out non-applicable):

* [x] Changes documented in `CHANGES.md`
* [x] Related issue exists and is referred to in the PR description and `CHANGES.md`
~* [ ] Added docstrings and API docs for any new/modified user-facing classes and functions~ n/a
* [x] Changes/features documented in `docs/*`
~* [ ] Unit-tests adapted/added for changes/features~ n/a
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: docs/start.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ version from PyPI:
 pip install zappend
 ```
 
+You can also install `zappend[dev]` to also install development dependencies,
```
Comment (thomasstorm) [inline docs/start.md:12]: I tried to come up with a good suggestion to get rid of the numerous "also install"s in this sentence, but couldn't think of any. Maybe you can rephrase anyway?

**Review by thomasstorm — APPROVED** [review]
Looks good! One tiny, non-crucial comment.

File: CHANGES.md [hunk 2]
```diff
@@ -1,7 +1,10 @@
-## Version 0.7.0 (in development)
+## Version 0.7.1 (in development)
 
 * Fixed link to _slice sources_ in documentation main page.
 
+* Moved all project configuration to `pyproject.toml` and removed
+  `setup.cfg` and requirements files.
```
Comment (forman) [inline CHANGES.md:6]:
```suggestion
  `setup.cfg` and requirements files. (#88)
```

**Review by forman — COMMENTED** [review]
Please merge my suggestion. That's all. Thanks, great!

File: docs/start.md (hunk 1 above)
Comment (pont-us) [reply docs/start.md]: I managed to get rid of one of them, but I'm keeping the other two (despite ugliness) for maximum clarity.

**Review by TejasMorbagal — APPROVED** [review]

**Review by TonioF — APPROVED** [review]
Looks good, Out of curiosity, why extend-ignore = E203 in .flake8?

**Pull request merged**